airflow/sensors/external_task.py

```diff
@@ -180,11 +198,20 @@ def poke(self, context, session=None):
 
         if count_failed == len(dttm_filter):
             if self.external_task_ids:
+                if self.soft_fail:
+                    raise AirflowSkipException(
+                        f'Some of the external tasks {self.external_task_ids} '
+                        f'in DAG {self.external_dag_id} failed. Skipping due to soft_fail.'
+                    )
                 raise AirflowException(
                     f'Some of the external tasks {self.external_task_ids} '
                     f'in DAG {self.external_dag_id} failed.'
                 )
             else:
+                if self.soft_fail:
+                    raise AirflowSkipException(
+                        f'The external DAG {self.external_dag_id} failed. ' 'Skipping due to soft_fail'
```

Comment:
```suggestion
                        f'The external DAG {self.external_dag_id} failed. Skipping due to soft_fail.'
```
Cleaning up a little implicit string concatenation.

Reply:
Ah sorry - this is me running black and not properly vetting the result. Will fix.